bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    @Nullable
+    @Override
+    public ThingUID getThingUID(@Nullable ServiceInfo service) {
+        logger.debug("ServiceInfo {}", service);
+        Validate.notNull(service);
+        return getThingUID(service.getName().toLowerCase(), "");
+    }
+
+    @Nullable
+    public ThingUID getThingUID(String name, String mode) {
+        String devid = StringUtils.substringAfterLast(name, "-");
+
+        if (name.startsWith("shelly1pm")) {
+            return new ThingUID(THING_TYPE_SHELLY1PM, devid);
+        }
+        if (name.startsWith("shellyem")) {
+            return new ThingUID(THING_TYPE_SHELLYEM, devid);
+        }
+        if (name.startsWith("shelly1")) {
+            return new ThingUID(THING_TYPE_SHELLY1, devid);
+        }
+        if (name.startsWith("shellyswitch25")) { // Shelly v2.5
+            if (mode.equals(SHELLY_MODE_RELAY)) {
+                return new ThingUID(THING_TYPE_SHELLY25_RELAY, devid);
+            }
+            if (mode.equals(SHELLY_MODE_ROLLER)) {
+                return new ThingUID(THING_TYPE_SHELLY25_ROLLER, devid);
+            }
+        }
+        if (name.startsWith("shellyswitch")) { // Shelly v2
+            if (mode.equals(SHELLY_MODE_RELAY)) {
+                return new ThingUID(THING_TYPE_SHELLY2_RELAY, devid);
+            }
+            if (mode.equals(SHELLY_MODE_ROLLER)) {
+                return new ThingUID(THING_TYPE_SHELLY2_ROLLER, devid);
+            }
+        }
+        if (name.startsWith("shelly4pro")) {
+            return new ThingUID(THING_TYPE_SHELLY4PRO, devid);
+        }
+        if (name.startsWith("shellyplug-s")) {
+            return new ThingUID(THING_TYPE_SHELLYPLUGS, devid);
+        }
+        if (name.startsWith("shellyplug")) {
+            return new ThingUID(THING_TYPE_SHELLYPLUG, devid);
+        }
+        if (name.startsWith("shellybulb")) {
+            return new ThingUID(THING_TYPE_SHELLYBULB, devid);
+        }
+        if (name.startsWith("shellysense")) {
+            return new ThingUID(THING_TYPE_SHELLYSENSE, devid);
+        }
+        if (name.startsWith("shellyht")) {
+            return new ThingUID(THING_TYPE_SHELLYHT, devid);
+        }
+        if (name.startsWith("shellysmoke")) {
+            return new ThingUID(THING_TYPE_SHELLYSMOKE, devid);
+        }
+        if (name.startsWith("shellyflood")) {
+            return new ThingUID(THING_TYPE_SHELLYFLOOD, devid);
+        }
+        if (name.startsWith("shellyrgbw2")) {
+            if (mode.equals(SHELLY_MODE_COLOR)) {
+                return new ThingUID(THING_TYPE_SHELLYRGBW2_COLOR, devid);
+            }
+            if (mode.equals(SHELLY_MODE_WHITE)) {
+                return new ThingUID(THING_TYPE_SHELLYRGBW2_WHITE, devid);
+            }
+        }
+        if (name.startsWith("shellydimmer")) {
+            return new ThingUID(THING_TYPE_SHELLYDIMMER, devid);
+        }
+
+        logger.info("Unsupported Shelly Device discovered: {} (mode {})", name, mode);
```

Comment:
Reduce to DEBUG

Reply:
info makes sense, otherwise the user never gets that info and will try to discover again and again